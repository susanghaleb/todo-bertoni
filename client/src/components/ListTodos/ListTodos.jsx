import React, { memo } from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";

import Todo from "../Todo/Todo";

const ListTodos = ({ tasks, onDelete, onCheck }) => {
  const useStyles = makeStyles(() => ({
    root: {
      width: "100%",
      justifyContent: "center"
    },
    inline: {
      display: "inline"
    }
  }));

  const classes = useStyles();
  return (
    <List className={classes.root}>
      {tasks.map(task => (
        <Todo
          task={task}
          onDelete={onDelete}
          onCheck={onCheck}
          key={JSON.stringify(task)}
        />
      ))}
    </List>
  );
};

ListTodos.propTypes = {
  tasks: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      id: PropTypes.number,
      done: PropTypes.bool
    })
  ),
  onDelete: PropTypes.func,
  onCheck: PropTypes.func
};

ListTodos.defaultProps = {
  tasks: [],
  onDelete: () => {},
  onCheck: () => {}
};

export default memo(ListTodos);
