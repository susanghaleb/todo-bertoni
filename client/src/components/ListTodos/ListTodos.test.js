import React from "react";
import renderer from "react-test-renderer";
import ListTodos from "./ListTodos";

it("renders correctly", () => {
  const mockFunction = jest.fn();
  const tree = renderer
    .create(
      <ListTodos tasks={[]} onCheck={mockFunction} onDelete={mockFunction} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
