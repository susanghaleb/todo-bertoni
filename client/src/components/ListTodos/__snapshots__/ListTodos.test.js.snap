// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="MuiList-root MuiList-padding makeStyles-root-1"
/>
`;
